src/test/java/uk/ac/ebi/eva/test/configuration/BatchTestConfiguration.java

```diff
@@ -15,8 +20,16 @@
 public class BatchTestConfiguration extends BaseTestConfiguration {
 
     @Bean
+    @ConditionalOnBean(value = {Job.class})
     public JobLauncherTestUtils jobLauncherTestUtils() {
         return new JobLauncherTestUtils();
     }
 
+    @Bean
+    @ConditionalOnBean(value = {Step.class})
+    @ConditionalOnMissingBean(value = {Job.class})
```

Comment:
Yes, they are totally required. This makes that if no job is found in the configuration of the test then we don't inject the JobLauncherTestUtils which would fail if no Job is found (Or it will throw a dependency injection error) , and then the StepLauncher would not be injected if no Step is found (Or it will throw a dependency injection error) or if a Job is found the context in which case you want to use the JobLauncherTestUtils not the StepLauncherTestUtils